R/plotAbundance.R

```diff
@@ -164,36 +163,36 @@ setMethod("plotAbundance", signature = c("SummarizedExperiment"),
             stop("No data to plot. nrow(x) == 0L.", call. = FALSE)
         }
         .check_assay_present(assay.type, x)
-        # if rank is set to NULL, default to plotExpression 
-        if(is.null(rank)){
-            plot <- plotExpression(x, features = features, 
-                                exprs_values = assay.type,
-                                one_facet = one_facet,
-                                ncol = ncol, scales = scales, ...)
-            ylab <- gsub("Expression","Abundance",plot$labels$y)
-            plot <- plot +
-                ylab(ylab)
-            return(plot)
-        }
         ############################# INPUT CHECK #############################
-        if(!.is_non_empty_string(rank)){
+        if(!.is_non_empty_string(rank) & !is.null(rank)){
             stop("'rank' must be an non empty single character value.",
                 call. = FALSE)
         }
         if(!.is_a_bool(use_relative)){
             stop("'use_relative' must be TRUE or FALSE.",
                 call. = FALSE)
         }
-        .check_taxonomic_rank(rank, x)
+        if(!is.null(rank)){
+            .check_taxonomic_rank(rank, x)
+        }
         .check_for_taxonomic_data_order(x)
         layout <- match.arg(layout, c("bar","point"))
         order_rank_by <- match.arg(order_rank_by, c("name","abund","revabund"))
         .check_abund_plot_args(one_facet = one_facet,
                             ncol = ncol)
-        if( !is.null(features) ){
+        if( !is.null(features) & !is.null(rank) ){
             features <- match.arg(features, colnames(colData(x)))
         }
+        if(is.null(rank) & ! is.null(order_sample_by)){
+            stop("'order_sample_by' must be set to null when 'rank' 
+                 is not specified.",
+                 call. = FALSE)
+        }
         ########################### INPUT CHECK END ###########################
+        if(is.null(rank)){
+           rowData(x)$rowNames <- row.names(rowData(x)) 
```

Comment:
row.names() might be better is some places. However, harmonized coding style is more important --> replace with rownames()


We can think about if that is something that we want to change from everywhere.

Also, you could rename this. It does not look good in legend of plot... How about "Feature"?

Moreover, just for a tip. It is always  better if you can reduce the number of places where you have hardcoded value. Here you have done it 2 times which means that if you modify just the other one, then the code does not work anymore. So tidier solution would be something like this:

```
rank <- "Feature"
rowData(x)[[ rank ]] <- rownames(x)
```